src/main/kotlin/apply/ui/admin/MenuItems.kt

```diff
@@ -0,0 +1,52 @@
+package apply.ui.admin
+
+import apply.application.RecruitmentResponse
+import com.vaadin.flow.component.Component
+import com.vaadin.flow.component.accordion.Accordion
+import com.vaadin.flow.component.details.DetailsVariant
+import com.vaadin.flow.component.html.Anchor
+import com.vaadin.flow.component.tabs.Tab
+import com.vaadin.flow.router.RouterLink
+import support.views.createTabs
+
+infix fun String.of(navigationTarget: Class<out Component>): MenuItem {
+    return SingleMenuItem(this, navigationTarget)
+}
+
+fun String.accordionOf(path: String, recruitments: List<RecruitmentResponse>): MenuItem {
+    return AccordionMenuItem(this, path, recruitments.map { it.toContent() })
+}
+
+private fun RecruitmentResponse.toContent(): AccordionContent = AccordionContent(id, title)
+
+sealed class MenuItem(protected val title: String) {
+    abstract fun toComponent(): Component
+}
```

Comment:
```suggestion
sealed class MenuItem {
    protected abstract val title: String
    abstract fun toComponent(): Component
}
```

프로퍼티 자체를 자식 클래스에서 재정의하게 해도 괜찮을 거 같아요.
자식 클래스에서 `title`이란 프로퍼티가 있는지 확실히 알 수 있는 점이 장점이겠네요.